### ISSUE TITLE
Multiple LoadBalancer source ranges not supported in Azure

### ISSUE THREAD

**Issue by karataliu** [kubernetes/kubernetes#43691]
**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
LoadBalancer, loadBalancerSourceRanges

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.4", GitCommit:"7243c69eb523aa4377bce883e7c0dd76b84709a1", GitTreeState:"clean", BuildDate:"2017-03-07T23:53:09Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: Azure ACS
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04 LTS (Xenial Xerus)
- **Kernel** (e.g. `uname -a`): Linux k8s-master-B09C34D8-0 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**What happened**:
When specifing multiple entries under 'loadBalancerSourceRanges', only the first one applies.

**What you expected to happen**:
All entries under 'loadBalancerSourceRanges' should apply, i.e. they should all appear in inbound security rules of cluster's Azure Network Security Group.

**How to reproduce it** (as minimally and precisely as possible):
Use kubectl to create a new service from a YAML config file, containing following section.
```
 loadBalancerSourceRanges:
  - 1.1.1.1/24
  - 2.1.1.1/24
```

After it reported completion, only '1.1.1.1/24' would appear in inbound security rules of cluster's Azure Network Security Group.

**Anything else we need to know**:

The issue lies here:
1. Same name is given for all the rules used by different loadBalancerSourceRanges entries
https://github.com/kubernetes/kubernetes/blob/90a382864e3650dfc0d1ae5535406cd93d7d077e/pkg/cloudprovider/providers/azure/azure_loadbalancer.go#L520

2. Later when it comes to the selecting process, all rules except the 1st one will be dropped, due to they share the same name
https://github.com/kubernetes/kubernetes/blob/90a382864e3650dfc0d1ae5535406cd93d7d077e/pkg/cloudprovider/providers/azure/azure_loadbalancer.go#L561

3. Later when the request was sent to Azure, only the first entry is included, resulting only the first one takes effect.

The problem is that current implementation is using the rule name to uniquely identify a rule, but the source range information is not captured by that name, which has the format: 'lbname-protocol-port-nodeport'.

We may need to either design new rule naming regulation to incorporate all required information, or update reconciliation logic for rule comparsion.

Also, https://github.com/kubernetes/kubernetes/issues/43608 should be with same cause, since the new updated rule uses same name, it won't get updated.

**Comment by IainColledge** [comment]
Is there a workaround for this on Azure you're aware of please?

**Comment by karataliu** [comment]
Related PR : https://github.com/kubernetes/kubernetes/pull/45523

**Comment by brendandburns** [comment]
Closing since this was fixed by #45523
